apps/air-discount-scheme/backend/src/app/modules/flight/flight.service.ts

```diff
@@ -62,6 +67,115 @@ export class FlightService {
+      !REYKJAVIK_FLIGHT_CODES.includes(secondFlight.destination) &&
+      !REYKJAVIK_FLIGHT_CODES.includes(secondFlight.origin)
+    ) {
+      return false
+    }
+
+    // If neither flight is connected to the allowed connecting flight places
+    // then it is not eligible
+    if (
+      !ALLOWED_CONNECTING_FLIGHT_CODES.includes(firstFlight.destination) &&
+      !ALLOWED_CONNECTING_FLIGHT_CODES.includes(firstFlight.origin) &&
+      !ALLOWED_CONNECTING_FLIGHT_CODES.includes(secondFlight.destination) &&
+      !ALLOWED_CONNECTING_FLIGHT_CODES.includes(secondFlight.origin)
+    ) {
+      return false
+    }
+
+    // Both flights need to touch Akureyri in some way, that is to say,
+    // Akureyri has to be a common point, ex: Reykjavik-Akureyri > Akureyri-Grimsey
+    // Logic: If not (Akureyri in destination or origin of strictly both flights) return false
+    if (
+      !(
+        AKUREYRI_FLIGHT_CODES.includes(firstFlight.destination) ||
+        AKUREYRI_FLIGHT_CODES.includes(firstFlight.origin)
+      ) ||
+      !(
+        AKUREYRI_FLIGHT_CODES.includes(secondFlight.destination) ||
+        AKUREYRI_FLIGHT_CODES.includes(secondFlight.origin)
+      )
+    ) {
+      return false
+    }
+
+    let delta = secondFlight.date.getTime() - firstFlight.date.getTime()
+
+    // The order must be flipped if we subtract the first intended chronological leg
+    // from the second intended chronological leg
+    if (
+      REYKJAVIK_FLIGHT_CODES.includes(secondFlight.origin) ||
+      REYKJAVIK_FLIGHT_CODES.includes(firstFlight.destination)
+    ) {
+      delta = -delta
+    }
+
+    if (delta >= 0 && delta <= CONNECTING_FLIGHT_GRACE_PERIOD) {
+      return true
+    }
+    return false
+  }
+
+  async isFlightLegConnectingFlight(
+    existingFlightId: string,
+    incomingLeg: FlightLeg,
+  ): Promise<boolean> {
+    // Get the corresponding flight for the connection discount code
+    const existingFlight = await this.flightModel.findOne({
+      where: {
+        id: existingFlightId,
+      },
+      include: [
+        {
+          model: this.flightLegModel,
+          where: { financialState: availableFinancialStates },
+        },
+      ],
+    })
+
+    if (!existingFlight) {
+      return false
+    }
+
+    // If a user flightLeg exists such that the incoming flightLeg makes a valid connection
+    // pair, return true
+    for (const flightLeg of existingFlight.flightLegs) {
+      if (
+        this.hasConnectingFlightPotentialFromFlightLegs(flightLeg, incomingLeg)
+      ) {
+        return true
+      }
+    }
+
+    return false
+  }
+
+  async findThisYearsConnectableFlightsByNationalId(
+    nationalId: string,
+  ): Promise<Flight[]> {
+    const flights = await this.findThisYearsFlightsByNationalId(nationalId)
+    // Filter out non-Reykjavík flights
+    return flights.filter((flight) => {
+      return (
+        flight.connectable &&
+        (REYKJAVIK_FLIGHT_CODES.includes(flight.flightLegs[0].origin) ||
```

Comment:
You should check here also if flightlegs connect to akureyri

Reply:
Yuss :+1:
Good point